### PR TITLE
Fixed a bug, where the tracker would stop automatic dispatching.

### PR DESCRIPTION
**Pull request by brototyp**

### CODE REVIEW

**Comment by minitrue22** [comment]
#167

File: PiwikTracker/PiwikTracker.swift [hunk 1]
```diff
@@ -96,6 +96,7 @@ final public class PiwikTracker: NSObject {
             guard events.count > 0 else {
                 // there are no more events queued, finish dispatching
                 self.isDispatching = false
+                self.startDispatchTimer()
```
Comment (minitrue22) [inline PiwikTracker/PiwikTracker.swift]: `startDispatchTimer` is called from different levels/abstractions in the code.
Eventually it might become hard to make sure all code branches are covered since the call is spread out in the code.

One option would be if `dispatchBatch()` took a closure that is called when it is done. Then the logic can be confined to the `dispatch()` method (I think).

File: PiwikTracker/PiwikTracker.swift [hunk 2]
```diff
@@ -96,6 +96,7 @@ final public class PiwikTracker: NSObject {
             guard events.count > 0 else {
                 // there are no more events queued, finish dispatching
                 self.isDispatching = false
```
Comment (minitrue22) [inline PiwikTracker/PiwikTracker.swift]: Just noticed that `isDispatching` is not thread safe, depending on what thread the caller calls the `dispatch()` method. There is a risk that the timer is not started/stopped as it should.

Comment (minitrue22) [inline PiwikTracker/PiwikTracker.swift]: I would fix that in a separate PR since the current bug is critical

File: PiwikTracker/PiwikTracker.swift [hunk 3]
```diff
@@ -105,15 +106,20 @@ final public class PiwikTracker: NSObject {
                     self.dispatchBatch()
                 })
             }, failure: { shouldContinue in
-                self.logger.warning("Failed dispatching events with error \(shouldContinue)")
```
Comment (thorstenstark) [inline PiwikTracker/PiwikTracker.swift]: Not part of this PR but what is `shouldContinue`? From the name I would say it's a `Bool`, but it is logged in the warning as error?

Comment (brototyp) [inline PiwikTracker/PiwikTracker.swift]: Thanks for that. I just renamed it.

File: PiwikTracker/PiwikTracker.swift (hunk 1 above)
Comment (brototyp) [inline PiwikTracker/PiwikTracker.swift]: Maybe it is possible to cleanup the whole dispatching process. It looks a bit overly complex for what it does. I will try to create a proposal.
